cardano-cli/src/Cardano/CLI/EraBased/Run/Query.hs

```diff
@@ -184,23 +186,39 @@ runQueryProtocolParametersCmd
   AnyCardanoEra era <- firstExceptT QueryCmdAcquireFailure $ newExceptT $ determineEra localNodeConnInfo
   sbe <- forEraInEon @ShelleyBasedEra era (left QueryCmdByronEra) pure
   let qInMode = QueryInEra $ QueryInShelleyBasedEra sbe Api.QueryProtocolParameters
-  pp <- firstExceptT QueryCmdConvenienceError
-          . newExceptT $ executeQueryAnyMode localNodeConnInfo qInMode
-  writeProtocolParameters sbe mOutFile pp
+  pParams <- firstExceptT QueryCmdConvenienceError
+              . newExceptT $ executeQueryAnyMode localNodeConnInfo qInMode
+  writeProtocolParameters sbe mOutFile pParams
   where
-    -- TODO: Conway era - use ledger PParams JSON
     writeProtocolParameters
       :: ShelleyBasedEra era
       -> Maybe (File () Out)
       -> Ledger.PParams (ShelleyLedgerEra era)
       -> ExceptT QueryCmdError IO ()
     writeProtocolParameters sbe mOutFile' pparams =
-      let apiPParamsJSON = (encodePretty $ fromLedgerPParams sbe pparams)
```

Comment:
I think we want to keep this `fromLedgerPParams` and do the patching on the cardano-api encoding of `ProtocolParameters`.